src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
@Aatch the name suggests that function goes ahead and (builds?) *checks* for overflow, when in reality it only checks whether overflow checks are enabled.

> Given the context of its use

I initially assumed this would already generate some code for overflow checking, perhaps, and was confused about the lack of arguments to it.

The fix could be as simple as making this `overflow_checked`, or, alternatively, inlined into `is_checkable`.